warehouse/slave.go

```diff
@@ -352,9 +374,20 @@ func processStagingFile(job PayloadT, workerIndex int) (loadFileUploadOutputs []
 	jobRun.setStagingFileDownloadPath(workerIndex)
 
 	// This creates the file, so on successful creation remove it
-	err = jobRun.downloadStagingFile()
+	err = jobRun.downloadStagingFile(jobRun.job.CurrentDestinationConfig, job.CurrentUseRudderStorage)
```

Comment:
Ideally a jobRun owns all the data necessary for performing operations around staging file. I was thinking that why we need to expose the logic to the caller by creating an opening for ability to supply the current destination config and use rudder storage options. Should we encapsulate it in the function itself ?

Reply:
Wanted to hear your thoughts on it ?

Reply:
Done.